pkg/controllers/cronfederatedhpa/cronfederatedhpa_job.go

```diff
@@ -307,24 +309,28 @@ func (c *ScalingJob) addSuccessExecutionHistory(
 		return exists
 	}
 
-	return retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
-		// If this history not exist, it means the rule deleted, so just ignore it.
-		if exists := addSuccessHistoryFunc(); !exists {
+	var operationResult controllerutil.OperationResult
+	if err := retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
+		var exists bool
+		operationResult, err = helper.UpdateStatus(context.Background(), c.client, cronFHPA, func() error {
+			// If this history not exist, it means the rule deleted, so just ignore it.
+			if exists = addSuccessHistoryFunc(); !exists {
+				return errors.New("no such rule")
+			}
 			return nil
-		}
-
-		updateErr := c.client.Status().Update(context.Background(), cronFHPA)
```

Comment:
> I found this problem in the production environment. 

Are you referring to this update conflict? 

> this modification can significantly reduce the probability of conflicts.

How do you explain the decrease?

Reply:
I don't know if you have read the [conversation](https://github.com/karmada-io/karmada/pull/4986#discussion_r1615410046) between me and @chaunceyjiang , which clearly explains why this PR is needed. As for how to measure whether the probability of conflict is reduced, to be honest, it is difficult for us to determine it through simple experiments, because conflict itself is a low-probability event.

Reply:
Oh, sorry. I get it.